base_location/tests/test_base_location.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2015 Yannick Vaucher, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests import tagged, common
+from odoo.tools.misc import mute_logger
+import psycopg2
+
+
+@tagged('post_install', '-at_install')
+class TestBaseLocation(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.state_vd = cls.env['res.country.state'].create({
```

Comment:
@aitorbouzas here `res.country.state` use multiple time, so you should you use global variable.

Reply:
I guess you mean to save the ref in a method-scope variable, right?

Reply:
@aitorbouzas just to clarify: `s/should/could` => is NOT mandatory.
It's actually a matter of taste + "ease creation" of other records w/out making the code redundant.
For instance: in my personal code I prefer that way as I'm lazy and I hate to copy/paste code :stuck_out_tongue:
but as you've seen I did not complain :wink: 
So all in all: nice to have but up to you.

Reply:
@simahawk Oh well, I know that I must have my own criteria for this type of things. Thanks :)

Reply:
> I guess you mean to save the ref in a method-scope variable, right?

Yes :+1: